src/main/java/net/glowstone/GlowServer.java

```diff
@@ -1090,7 +1095,11 @@ public void setPlayerOnline(GlowPlayer player, boolean online) {
 
     @Override
     public String getName() {
-        return getClass().getPackage().getImplementationTitle();
+        if (isNameNull)   {
```

Comment:
Again, I'm so sorry for not being specific. I meant store the name within the method. Though, what you did right now might actually be better. I'll test and see.